Source/Particles/Pusher/ImplicitPushPX.cpp

```diff
@@ -226,6 +226,17 @@ namespace {
 
         }
 
+        // The Charge-conserving deposits assume the change in position is consistent with
```

Comment:
Instead of doing this separate push here, could the `else if` be added to the convergence check above
```
if (step_norm < particle_tolerance) {
    convergence = true;
    break;
} else if (iter+1 == max_iterations) {
    break;
}
```
The code would then need to set `options.max_particle_iterations = 2` instead of `1`.

Reply:
My main concern is that this is duplicated code. Also, I would find it more clear what is happening seeing that break after the position advance, that the position advance always needs to happen so that the positions are consistent with the velocities.

I think this could be made more clear if the loop was changed to a `while (true)` loop so that the only exit points are right after the position advance.